lib/cli/src/Cardano/CLI.hs

```diff
@@ -570,6 +564,7 @@ data TransactionListArgs = TransactionListArgs
     , _walletId :: WalletId
     , _timeRangeStart :: Maybe Iso8601Time
     , _timeRangeEnd :: Maybe Iso8601Time
+    , _sortOrder :: Maybe SortOrder
```

Comment:
@Anviking wrote:

> Would `direction` be a more precise term?
> My impression was that **order** refers to all possible `a -> a -> Ordering` (`Ordering = EQ, GT, LT`) implementations, whereas **direction** only specifies whether that function should be inverted or not.

I think `SortDirection` could certainly also be acceptable.

However in that case, it would be probably be more natural to say "forward direction" and "reverse direction". (The terms "ascending direction" and "descending direction" sound quite unnatural to my ears.)

In my experience, "ascending order" and "descending order" are already commonly-used noun phrases, with well-defined meanings. For example, consider this usage (emphasis mine):

> "A **standard order** is often called **ascending** (corresponding to the fact that the standard order of numbers is ascending, i.e. A to Z, 0 to 9), the **reverse order** is often called **descending** (Z to A, 9 to 0). For dates and times, ascending means that earlier values precede later ones e.g. 1/1/2000 will sort ahead of 1/1/2001."
([source](https://en.wikipedia.org/wiki/Sorting))

Regarding:

> While on the topic, I think `since` and `until` as used in the [commits api](https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository) are nice alternatives to `start` and `end`, but might not work well when used with `descending` order.

As I see it, the problem with the word "since" is that it has an _implied directionality_ that goes _forward in time_. But the words `start` and `end` simply define boundaries that could be anywhere in time. They could even be reversed, and it wouldn't contradict the English meanings. Consider this made-up example:

> "In order to produce the descending list of integers from 10 to 1, we **start** at 10, apply the `pred` function repeatedly, and stop when we reach the **end**, at 1".

If we try to substitute the word "since" here, then it doesn't work as nicely.